src/main/java/net/minecraftforge/common/hunger/FoodValues.java

```diff
@@ -0,0 +1,128 @@
+{
+    /**
+     * @return The amount of saturation that the food values would provide, ignoring any limits.
+     */
+    public float getUnboundedSaturationIncrement()
+    {
+        return nutrition * saturationModifier * 2f;
+    }
+
+    /**
+     * @return The bounded amount of saturation that the food values would provide to this player,
+     * taking their max hunger level into account.
+     */
+    public float getSaturationIncrement(Player player)
+    {
+        return Math.min(ForgeEventFactory.getMaxHunger(player), getUnboundedSaturationIncrement());
+    }
+
+    /**
+     * Get unmodified (vanilla) food values.
+     *
+     * @return The food values, or null if none were found.
+     */
+    public static FoodValues getUnmodified(@Nonnull ItemStack stack)
+    {
+        if (stack != ItemStack.EMPTY)
+        {
+            if (stack.isEdible())
+            {
+                FoodProperties food = stack.getItem().getFoodProperties();
+                return new FoodValues(food.getNutrition(), food.getSaturationModifier());
+            }
+            else if (stack.getItem() instanceof IEdible edible)
+            {
+                return edible.getFoodValues(stack);
+            }
+            else if (stack.getItem() instanceof BlockItem blockItem)
+            {
+                if (blockItem.getBlock() instanceof IEdible edible)
+                {
+                    return edible.getFoodValues(stack);
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Get player-agnostic food values.
+     *
+     * @return The food values, or null if none were found.
+     */
+    public static FoodValues get(@Nonnull ItemStack stack)
+    {
+        return get(stack, null);
+    }
+
+    /**
+     * Get player-specific food values.
+     *
+     * @return The food values, or null if none were found.
+     */
+    public static FoodValues get(@Nonnull ItemStack stack, @Nullable Player player)
+    {
+        FoodValues foodValues = getUnmodified(stack);
+        if (foodValues != null)
+        {
+            return ForgeEventFactory.getFoodValues(foodValues, stack, player);
+        }
+        return null;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + nutrition;
+        result = prime * result + Float.floatToIntBits(saturationModifier);
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        FoodValues other = (FoodValues) obj;
+        if (nutrition != other.nutrition)
+            return false;
+        if (Float.floatToIntBits(saturationModifier) != Float.floatToIntBits(other.saturationModifier))
+            return false;
+        return true;
+    }
```

Comment:
These are automatically implemented for record classes based on the fields.  Technically fine to keep, but unnecessary.

Reply:
Huh...I was pretty sure I removed these...but I guess not! Thanks for double-checking.